autogen/README.md

```diff
@@ -0,0 +1,86 @@
+# Autogen Framework Model Interfaces
+The Autogen framework gives 3 major file based interfaces for models: `model.json`, `modelparameters.m`, and `sm_stop_verify.m` and 1 major interface available as a struct in the Matlab workspace: `mp`.
+
+## model.json
+`model.json` is a required configuration file shared between the Autogen Framework and the Autogen Tools. This covers both system-level properties as well as run-time tunable parameters for the model via registers.  
+`model.json` properties:  
+- `system`: An object to describe the overall system  
+    - `target`: Which hardware system is being targeted, the `audiomini` or `audioblade` (default: `audiomini`)  
+    - `sampleClockFrequency`: The sampling rate of the audio being processed (default: `48000`)  
+    - `systemClockFrequency`: Clock rate of the audio processing on the FPGA (default: `98304000`)  
```

Comment:
I currently have defaults set up currently for some `model.json` attributes like `systemClockFrequency`. What that means right now is that the field is actually optional in `model.json` because the framework will use the default. The downside to this is it means everything downstream in `simulink_codgen` needs to implement matching defaults.

What are thoughts making a small script instead that generates `model.json` with the defaults so that way we don't to support missing fields in a variety of areas?

Reply:
Having optional fields makes making `model.json` by hand less cumbersome, but I'm not a huge fan of needing checks for missing optional values in multiple places downstream, as that seems problematic if we decide to change default values in the future. 

Would the script that generates `model.json` prompt the user for input and set defaults if the user just hits enter? I think that would be good. Another option would be to have the simulink framework insert missing default values in the json before it gets used downstream. Of those two options, I prefer the `model.json` generator script.

Reply:
The simplest version script would only take a device/model name, everything else is autofilled with the defaults. It definitely could let them set values on everything, but I don't see it as overly needed because it is essentially the same amount of work to type into our utility as it is to just modify the file.

Reply:
Yeah, there isn't particularly any time savings by prompting the user. If anything, there could be some minor annoyance if they have to keep hitting enter for defaults. 

Is there an advantage to having this script versus having users copy a default `model.json` and modifying it? I suppose it could be nice to have a script that not only makes the `model.json`, but that also provides all of the files needed to start developing a model. This would be a sort of "project initialization" script that creates a `model.json`; a template Simulink model with the appropriate name, etc.;  and any other files that typically live in the model directory, such as `modelparameters.m` and the verification script. Thinking towards a future where we have a Toolbox, this "project initialization" script could be tied into the toolbox, perhaps even graphically if we felt like it.

Reply:
While more could certainly be done, the simplest thing is that you don't have to find a file, which if this is in a Toolbox is even more important since they shouldn't need to look in an installed Toolbox's files for the template. Copy file from `/random/path/here` vs call `genModelJson("echo")`

Reply:
I'm in favor of creating a `genModelJson` or `createModelJson` function. Let's keep it simple as you suggest. Are you envisioning this as a python script so it can be used without Matlab?

Reply:
createModelJson has been created. I just take `defaultModel.json`, jsondecode it, set the device name, and dump it into the calling directory with writejson